### PR TITLE
initial update sort behavior

### PR DESCRIPTION
**Pull request by bgila**
https://linear.app/manifoldmarkets/issue/MAN-1376/market-creator-should-be-able-to-set-default-sort-default-value-for

discussed with ian

### CODE REVIEW

File: web/components/contract/contract-overview.tsx [hunk 1]
```diff
@@ -560,6 +565,24 @@ const ChoiceOverview = (props: {
...
   )
 
+  useEffect(() => {
+    if (
+      currentUserId &&
+      (isModId(currentUserId) ||
+        isAdminId(currentUserId) ||
+        contract.creatorId === currentUserId)
+    ) {
+      updateContract(contract.id, { sort }).then(
```
Comment (IanPhilips) [inline web/components/contract/contract-overview.tsx:582]: I think you can wrap this in a `toast.promise`

File: web/components/contract/contract-overview.tsx [hunk 2]
```diff
@@ -560,6 +565,24 @@ const ChoiceOverview = (props: {
     [answers, resolutions, shouldAnswersSumToOne, sort]
   )
 
+  useEffect(() => {
+    if (
+      currentUserId &&
```
Comment (IanPhilips) [inline web/components/contract/contract-overview.tsx:577]: I think this logic is cursed, should be `currentUserId && (case1 || case2 || case3)`

**Review by IanPhilips — CHANGES_REQUESTED** [review]
Almost there! Last thing you need to do is add the logic in the `/backend/api/src/update-market.ts`

**Comment by bgila** [comment]
addressed, ty

**Pull request merged**